serde-support/src/main/java/io/micronaut/serde/support/deserializers/DemuxingObjectDecoder.java

```diff
@@ -212,6 +220,15 @@ Decoder consume() {
                 }
             }
 
+            Decoder buffer() throws IOException {
```

Comment:
how can this work, you cant use the Decoder returned by decodeBuffer multiple times